File: server/src/main/java/by/brawl/util/Exceptions.java
package by.brawl.util;

import org.slf4j.Logger;
import org.springframework.security.access.AccessDeniedException;
import org.springframework.security.core.userdetails.UsernameNotFoundException;

import java.text.MessageFormat;

public final class Exceptions {

	private Exceptions() {
		// Lock class from instantiation
	}

	public static AccessDeniedException produceAccessDenied(Logger log, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		AccessDeniedException e = new AccessDeniedException(formattedMessage);
		log.error(formattedMessage, e);
		return e;
	}

	public static IllegalArgumentException produceIllegalArgument(Logger log, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		IllegalArgumentException e = new IllegalArgumentException(formattedMessage);
		log.error(formattedMessage, e);
		return e;
	}

	public static IllegalStateException produceIllegalState(Logger log, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		IllegalStateException e = new IllegalStateException(formattedMessage);
		log.error(formattedMessage, e);
		return e;
	}

	public static NullPointerException produceNullPointer(Logger log, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		NullPointerException e = new NullPointerException(formattedMessage);
		log.error(formattedMessage, e);
		return e;
	}

	public static UsernameNotFoundException produceUsernameNotFound(Logger log, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		UsernameNotFoundException e = new UsernameNotFoundException(formattedMessage);
		log.error(formattedMessage, e);
		return e;
	}

	public static void logError(Logger log, Exception e, String message, Object... messageArgs) {
		String formattedMessage = MessageFormat.format(message, messageArgs);
		log.error(formattedMessage, e);
	}
}

File: server/src/main/java/by/brawl/util/Exceptions.kt
package by.brawl.util

import org.slf4j.Logger
import org.springframework.security.access.AccessDeniedException
import org.springframework.security.core.userdetails.UsernameNotFoundException

object Exceptions {

    fun produceAccessDenied(log: Logger, message: String): AccessDeniedException {
        val e = AccessDeniedException(message)
        log.error(message, e)
        return e
    }

    fun produceIllegalArgument(log: Logger, message: String): IllegalArgumentException {
        val e = IllegalArgumentException(message)
        log.error(message, e)
        return e
    }

    fun produceIllegalState(log: Logger, message: String): IllegalStateException {
        val e = IllegalStateException(message)
        log.error(message, e)
        return e
    }

    fun produceNullPointer(log: Logger, message: String): NullPointerException {
        val e = NullPointerException(message)
        log.error(message, e)
        return e
    }

    fun produceUsernameNotFound(log: Logger, message: String): UsernameNotFoundException {
        val e = UsernameNotFoundException(message)
        log.error(message, e)
        return e
    }

    fun logError(log: Logger, e: Exception, message: String) {
        log.error(message, e)
    }

}